redisson/src/main/java/org/redisson/connection/balancer/LoadBalancerManager.java

```diff
@@ -148,9 +149,33 @@ protected void onSuccess(Void value) {
                         }
                     };
                     listener.setCounter(2);
-                    BiConsumer<Void, Throwable> initCallBack = (r, ex) -> {
-                        if (ex == null) {
-                            listener.decCounter();
+                    BiConsumer<Void, Throwable> initCallBack = new BiConsumer<Void, Throwable>() {
+                        private AtomicBoolean initConnError = new AtomicBoolean(false);
+                        @Override
+                        public void accept(Void r, Throwable ex) {
+                            synchronized (this) {
+                                if (ex == null) {
+                                    listener.decCounter();
+                                } else {
+                                    if (!initConnError.compareAndSet(false, true)) {
+                                        return;
+                                    }
+                                    for (RedisConnection connection : entry.getAllConnections()) {
```

Comment:
don't we do this already at https://github.com/redisson/redisson/blob/8c93a3c8860a89b4f218bcaae7bf67717a42c5b3/redisson/src/main/java/org/redisson/connection/pool/ConnectionPool.java#L135 ?